text/0000-javascript-module-api.md

```diff
@@ -0,0 +1,1131 @@
+| Module                                          | Global                   |
+| ---                                             | ---                      |
+| `import { begin } from "@ember/runloop"`        | `Ember.run.begin`        |
+| `import { bind } from "@ember/runloop"`         | `Ember.run.bind`         |
+| `import { cancel } from "@ember/runloop"`       | `Ember.run.cancel`       |
+| `import { debounce } from "@ember/runloop"`     | `Ember.run.debounce`     |
+| `import { end } from "@ember/runloop"`          | `Ember.run.end`          |
+| `import { join } from "@ember/runloop"`         | `Ember.run.join`         |
+| `import { later } from "@ember/runloop"`        | `Ember.run.later`        |
+| `import { next } from "@ember/runloop"`         | `Ember.run.next`         |
+| `import { once } from "@ember/runloop"`         | `Ember.run.once`         |
+| `import { run } from "@ember/runloop"`          | `Ember.run`              |
+| `import { schedule } from "@ember/runloop"`     | `Ember.run.schedule`     |
+| `import { scheduleOnce } from "@ember/runloop"` | `Ember.run.scheduleOnce` |
+| `import { throttle } from "@ember/runloop"`     | `Ember.run.throttle`     |
+
+### `@ember/service`
+| Module                                    | Global                 |
+| ---                                       | ---                    |
+| `import Service from "@ember/service"`    | `Ember.Service`        |
+| `import { inject } from "@ember/service"` | `Ember.inject.service` |
+
+### `@ember/string`
+| Module                                       | Global                    |
+| ---                                          | ---                       |
+| `import { camelize } from "@ember/string"`   | `Ember.String.camelize`   |
+| `import { capitalize } from "@ember/string"` | `Ember.String.capitalize` |
+| `import { classify } from "@ember/string"`   | `Ember.String.classify`   |
+| `import { dasherize } from "@ember/string"`  | `Ember.String.dasherize`  |
+| `import { decamelize } from "@ember/string"` | `Ember.String.decamelize` |
+| `import { fmt } from "@ember/string"`        | `Ember.String.fmt`        |
+| `import { htmlSafe } from "@ember/string"`   | `Ember.String.htmlSafe`   |
+| `import { loc } from "@ember/string"`        | `Ember.String.loc`        |
+| `import { underscore } from "@ember/string"` | `Ember.String.underscore` |
+| `import { w } from "@ember/string"`          | `Ember.String.w`          |
+
+### `@ember/utils`
+| Module                                     | Global            |
+| ---                                        | ---               |
+| `import { isBlank } from "@ember/utils"`   | `Ember.isBlank`   |
+| `import { isEmpty } from "@ember/utils"`   | `Ember.isEmpty`   |
+| `import { isNone } from "@ember/utils"`    | `Ember.isNone`    |
+| `import { isPresent } from "@ember/utils"` | `Ember.isPresent` |
+| `import { tryInvoke } from "@ember/utils"` | `Ember.tryInvoke` |
+| `import { type
```

Comment:
Perhaps export an install function to keep the module itself from having side effects?

In other words, something like:
```
import { install } from '@ember/extensions/string';

if (config.someFlag) {
   install();
}
```

This would allow installing the extension conditionally. It should be possible to configure compiler to eliminate the install + import at build time (like has-js style "tree shaking").

Prototype extensions are currently controlled by the environment settings. Where would this import go? The app module? Could an addon import this module and enable proto extensions for the entire app?

Perhaps folks will argue that this creates another step for users.